client/public/static/js/recipes/recipe_details.js

```diff
@@ -511,41 +533,103 @@ function handleUpdateRecipe() {
   window.location = `/account/update_recipe?userRecipeName=${recipeName}`;
 };
 
-async function handlePublishUserRecipe(userId,form) {
+async function handlePublishUserRecipe(userId, form, showUsername) {
   const recipeName = document.getElementById('recipe-name').textContent;
-  const buttonText = form.find("#publishRecipeBtn").text();
-  if(buttonText.includes("remove")){
```

Comment:
This remove should not have been taken out. Everything in this first if needs to stay. It is how the end user removes their published recipe.